test/discount_code_test.rb

```diff
@@ -0,0 +1,50 @@
+require 'test_helper'
+
+class DiscountCodeTest < Test::Unit::TestCase
+  def setup
+    super
+    fake 'price_rules/102586120/discount_codes/1002091923', body: load_fixture('discount_code')
+
+    @discount_code = ShopifyAPI::DiscountCode.find(1002091923, params: {price_rule_id: 102586120})
+  end
+
+  def test_get_discount_code
+    fake 'price_rules/102586120/discount_codes', method: :get, status: 200, body: load_fixture('discount_code')
+    discount_code = ShopifyAPI::DiscountCode.find 1002091923, params: { price_rule_id: 102586120 }
+
+    assert_equal 1002091923, discount_code.id
+  end
+
+  def test_get_all_discount_codes
+    fake 'price_rules/102586120/discount_codes', method: :get, status: 200, body: load_fixture('discount_codes')
+    discount_codes = ShopifyAPI::DiscountCode.all params: { price_rule_id: 102586120 }
+
+    assert_equal 1, discount_codes.length
+    assert_equal 1002091923, discount_codes.first.id
+  end
+
+  def test_create_discount_code
+    fake 'price_rules/102586120/discount_codes', method: :post, status: 201, body: load_fixture('discount_code')
+    discount_code = ShopifyAPI::DiscountCode.new
+    discount_code.prefix_options[:price_rule_id] = 102586120
```

Comment:
is that a common pattern? should this implementation detail be abstracted with a setter?

```ruby
module ShopifyAPI
  class DiscountCode < Base
    ...
    def price_rule_id=(id)
      @prefix_options[:price_rule_id] = id
    end
  end
end
...
discount_code.price_rule_id = 102586120
```